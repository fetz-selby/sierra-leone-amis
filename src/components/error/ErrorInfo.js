import React from 'react';
import PropTypes from 'prop-types';
import wifi from '../../assets/icons/no-con.svg';
import './error-info.css';

const ErrorInfo = (props)=>
    <div className='error-info-container'>
        <div className='icon-layout'>
            <img className='icon' src={wifi} alt='no connection' />
        </div>
        <div className='info-layout'>
            <div className='message-container'>
                <div className='message'>{props.message}</div>
            </div>
            <div className='action-container'>
                <div className='btn-link' onClick={props.onCancel}>close</div>
            </div>
        </div>
    </div>


ErrorInfo.propTypes = {
    message: PropTypes.string.isRequired,
    onCancel: PropTypes.func
}

export default ErrorInfo;