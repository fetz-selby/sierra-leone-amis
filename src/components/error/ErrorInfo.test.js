import React from 'react';
import {shallow} from 'enzyme';
import ErrorInfo from './';

describe('Network Error Component', ()=>{
    it('should display [message] passed with props', ()=>{
        const message = 'network is down';

        const error = shallow(<ErrorInfo message={message}/>);
        expect(error.find('.message').text()).toBe(message);
    })

    it('should call onCancel when cancel is clicked', ()=>{
        const message = 'network is down';
        const id = 1;
        const onCancel = jest.fn();

        const props = {
            message,
            onCancel
        }

        const error = shallow(<ErrorInfo {...props}/>);
        error.find('.btn-link').simulate('click');
        expect(onCancel).toHaveBeenCalled();
    })

    it('should render correctly', ()=>{
        const message = 'network is down';

        const props = {
            message
        }

        const error = shallow(<ErrorInfo {...props}/>);
        expect(error).toMatchSnapshot();
    })
})