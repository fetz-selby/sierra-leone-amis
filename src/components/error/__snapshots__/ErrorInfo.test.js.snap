// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Error Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorInfo
    message="network is down"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon-layout"
        >
          <img
            alt="no connection"
            className="icon"
            src="no-con.svg"
          />
        </div>,
        <div
          className="info-layout"
        >
          <div
            className="message-container"
          >
            <div
              className="message"
            >
              network is down
            </div>
          </div>
          <div
            className="action-container"
          >
            <div
              className="btn-link"
            >
              close
            </div>
          </div>
        </div>,
      ],
      "className": "error-info-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="no connection"
            className="icon"
            src="no-con.svg"
          />,
          "className": "icon-layout",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "no connection",
            "className": "icon",
            "src": "no-con.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="message-container"
            >
              <div
                className="message"
              >
                network is down
              </div>
            </div>,
            <div
              className="action-container"
            >
              <div
                className="btn-link"
              >
                close
              </div>
            </div>,
          ],
          "className": "info-layout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="message"
              >
                network is down
              </div>,
              "className": "message-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "network is down",
                "className": "message",
              },
              "ref": null,
              "rendered": "network is down",
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="btn-link"
              >
                close
              </div>,
              "className": "action-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "close",
                "className": "btn-link",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "close",
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-layout"
          >
            <img
              alt="no connection"
              className="icon"
              src="no-con.svg"
            />
          </div>,
          <div
            className="info-layout"
          >
            <div
              className="message-container"
            >
              <div
                className="message"
              >
                network is down
              </div>
            </div>
            <div
              className="action-container"
            >
              <div
                className="btn-link"
              >
                close
              </div>
            </div>
          </div>,
        ],
        "className": "error-info-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="no connection"
              className="icon"
              src="no-con.svg"
            />,
            "className": "icon-layout",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "no connection",
              "className": "icon",
              "src": "no-con.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="message-container"
              >
                <div
                  className="message"
                >
                  network is down
                </div>
              </div>,
              <div
                className="action-container"
              >
                <div
                  className="btn-link"
                >
                  close
                </div>
              </div>,
            ],
            "className": "info-layout",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="message"
                >
                  network is down
                </div>,
                "className": "message-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "network is down",
                  "className": "message",
                },
                "ref": null,
                "rendered": "network is down",
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn-link"
                >
                  close
                </div>,
                "className": "action-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "close",
                  "className": "btn-link",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "close",
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
